### PR TITLE
Aggregations

### PR DESCRIPTION
**Pull request by perolavsvendsen**
Rough first version draft of aggregated data support. Trying to follow the current setup more or less, so that functions hopefully can be used across after some refactoring. Not sure if good approach.

#59

### CODE REVIEW

**Comment by perolavsvendsen** [comment]
Waiting for feedback on this. Go further with this concept, or not?

File: examples/s/d/nn/_project/aggregate_surfaces.py
Comment (jcrivenaes) [inline examples/s/d/nn/_project/aggregate_surfaces.py:167]: Do we need kwarg `obj=surface`? or could we live with arg here as in `ExportData.export()`

`metadata = exp.generate_metadata(surface, operation=operation)`

File: examples/s/d/nn/_project/aggregate_surfaces.py
Comment (jcrivenaes) [inline examples/s/d/nn/_project/aggregate_surfaces.py:158]: Clarify what could happen

File: examples/s/d/nn/xcase/realization-0/iter-0/rms/bin/export_a_surface.py
Comment (jcrivenaes) [inline examples/s/d/nn/xcase/realization-0/iter-0/rms/bin/export_a_surface.py:60]: Workflow for depth is "rms structural model"

In this routine, make a loop in main and separate the export to a more generic function, where a few keys are different between a property and a depth (the point is avoid too much duplicate code)

**Review by jcrivenaes — CHANGES_REQUESTED** [review]
I have mostly looked at the "end-user contract API" from the examples, and I think it is on the right track. The question is how this fits with pre-processed/auxilary data. Can this function be reused here (which require class renaming). Or do we need additional classes?

I also propose a rebase to fewer commits, but where each commit has a clear purpose/task. Then it is easier to review a second time.

Last, a clear and concise end-user documentation (sphinx) shall be a separate PR, and I propose we make and merge that before any merging of this PR

**Comment by perolavsvendsen** [comment]
> I have mostly looked at the "end-user contract API" from the examples, and I think it is on the right track. The question is how this fits with pre-processed/auxilary data. Can this function be reused here (which require class renaming). Or do we need additional classes?

Possibly - I can try out some variations and see what it feels like. In that case, it would possibly be one class for the "realization" context (with RMS template runs as a special case) and one for the `pre` and `post` contexts.

> I also propose a rebase to fewer commits, but where each commit has a clear purpose/task. Then it is easier to review a second time.

Yes, reflecting lack of plan...

> Last, a clear and concise end-user documentation (sphinx) shall be a separate PR, and I propose we make and merge that before any merging of this PR

Yes, good input. I will have a look at that. I agree that documentation -> code is a good pattern.

File: examples/s/d/nn/_project/aggregate_surfaces.py
Comment (perolavsvendsen) [reply examples/s/d/nn/_project/aggregate_surfaces.py:167]: No, not needed. See #192

File: examples/s/d/nn/_project/aggregate_surfaces.py
Comment (perolavsvendsen) [reply examples/s/d/nn/_project/aggregate_surfaces.py:158]: Gone in #192

File: examples/s/d/nn/xcase/realization-0/iter-0/rms/bin/export_a_surface.py
Comment (perolavsvendsen) [reply examples/s/d/nn/xcase/realization-0/iter-0/rms/bin/export_a_surface.py:60]: > In this routine, make a loop in main and separate the export to a more generic function, where a few keys are different between a property and a depth (the point is avoid too much duplicate code)

Will involve quite a lot of config on the outside, and first attempts did not end up very readable compared to current. Problem is that this example is not really representative for actual usage. Normally, these would be in separate scripts, and each be >1 surface. Then it would make sense to loop. Can keep this open for discussion.

**Comment by perolavsvendsen** [comment]
Redundant after refactoring

**Pull request closed without merging**
